src/lsm/grid.zig

```diff
@@ -432,7 +425,27 @@ pub fn GridType(comptime Storage: type) type {
                 .completion = undefined,
                 .block = block,
             };
+
+            // Collect the current Read and any other matching/pending Reads to this IOP.
+            // There could be reads in the read_queue for the same address that our read
+            // finally got an IOP for. If we don't gather them here, they will eventually be 
+            // processed at the end of read_block_callback() but that will issue a new call to
+            // read_sectors().
             iop.reads.push(read);
```

Comment:
Add a comment to the `read_iops` field declaration of the form:

```
/// Invariants:
/// - ...
```

that describes how read iops refer to unique reads (because duplicate reads are chained together).